enhancements-1: caching, refactoring, styling, material-ui

codealchemist opened the pull request:
Added material-ui (next) with svg icons; moved order-day component to its own file; updated orders styling; added list view mode with view mode selector as its own component; highlighting current date; added roboto font; updated dates to get current date and updated menu_id on constants to work with current month (October); caching orders food list; running on port 9000 to avoid conflicts with services we already run on port 8000; added editorconfig.

codealchemist commented:
Current day highlighting in calendar view mode:
![image](https://user-images.githubusercontent.com/1118293/31681035-6a979610-b34c-11e7-8b4e-7089d19a2ba5.png)

Current day highlighting in list view mode:
![image](https://user-images.githubusercontent.com/1118293/31681067-8396e47c-b34c-11e7-961b-0edb23514f7c.png)

codealchemist commented:
View mode selector:
![image](https://user-images.githubusercontent.com/1118293/31681096-9f57ffc0-b34c-11e7-8c87-eaf15d198738.png)

santiac89 commented on src/services/orders.js:
```diff
@@ -11,18 +12,30 @@ me.get = () => {
 
 me.getMenuOptions = (date) => {
+  const data = localStorageSvc.get(date);
+  if (data) {
```
Seguro no se da mucho el caso, pero aca quedaria agregar algun tipo de invalidacion de cache 🤔

codealchemist replied:
Sí, habría que verlo.
Mi idea general fue que el primer día del mes puedo cargar el mes completo.
O sea que no debería cambiar el menu para cada día, o si cambia deberían respetar lo que se cargó.
Entonces sería safe cachear por fecha.
El único caso particular que veo es si permitimos ver el mes siguiente y aún no está habilitado y trae empty data o bad data, ahí habría que evitar llenar el cache.
Tenemos que ver bien cómo darnos cuenta si el mes siguiente está habilitado o no.

santiac89 commented on src/services/constants.js:
```diff
@@ -2,4 +2,4 @@ import md5 from 'md5';
 export const BASE_ENDPOINTS = '/chef-gourmet/public/ws';
 export const CHEF_PASS = md5('chefgourmet');
-export const MENU_ID = 20;
+export const MENU_ID = 21;
```
Esto al final es fijo?

codealchemist replied:
No, tenemos que ver cómo lo obtenemos.
Parece incrementar de a uno cada mes.
Pero note tendríamos que asegurar.

lcortese replied:
menu_id es el menu disponible para x día, si get-orders no retorna menu_id. no se puede setear la comida para ese día.

codealchemist replied:
Great finding @lcortese!
Looking forward to implement this.
We need to hit `get-orders` endpoint to get the `menu_id` for each day.
It will also tell us if we can make an order for that day or not.
👍

santiac89 commented on src/components/orders-page/view-mode.jsx:
```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
```
Quizas aprovechando que agregaste material-ui podemos reemplazar este componente por un simple switch?

codealchemist replied:
Yeah, puede ser!

codealchemist replied:
Tampoco me parece mal tener un componente para el view mode.
De hecho le falta agregar algunas cosas aún, como resaltar el current view mode.
Y hacer que no se pueda seleccionar el current.
Si luego agregamos algún otro view mode pinta más copado que esté en su propio component too.

santiac89 commented on src/components/orders-page/order-day.jsx:
```diff
@@ -0,0 +1,73 @@
+  };
+
+
+  me.loadMenuOptions(day);
```
Esto lo meteria en un componentDidMount o algun otro hook del lifecycle, a menos que sea por algo que esta aca

codealchemist replied:
Esto es code existente, no es parte del PR.
Tiene sentido hookearlo al lifecycle igualmente.

codealchemist commented on src/index.html:
```diff
@@ -6,7 +6,9 @@
     <meta http-equiv="Content-type" content="text/html; charset=utf-8"/>
     <title><%= htmlWebpackPlugin.options.title %></title>
+
+    <link href="https://fonts.googleapis.com/css?family=Roboto:300,400,500" rel="stylesheet">
```
Sí, ni hablar.

codealchemist commented:
Updated calendar view:
![image](https://user-images.githubusercontent.com/1118293/31752702-b678bf66-b462-11e7-80ba-cf7736c84b57.png)

Updated list view:
![image](https://user-images.githubusercontent.com/1118293/31752721-c9f2fd40-b462-11e7-812f-adcf3715bdbe.png)

codealchemist commented:
Now we display selected food for each day with a simple status to indicate if it's selected or not.
`menu_id` loaded dynamically for each day from `get-orders` response.
Displaying *NA* status for disabled days.

If you guys think we can continue enhancing it from here let's merge.
:)

Thx!

lcortese commented on src/components/orders-page/index.jsx:
```diff
@@ -1,106 +1,91 @@
-    days: []
+    days: [],
+    viewMode: 'calendar',
+    orders: []
```
De momento esta bien. Pero por experiencia siempre sugieron que la entidad sea un objeto con mas atributos. quiero decir:
```
me.state = {
	orders: {
		isLoading: false,
		data: [],
		pagination: {...}
	},
	...
};
```

codealchemist replied:
Sep, tenemos que manejar el loading state!
Y después vemos qué más.
Por ahora sólo veo eso que falta.
